data_exploration.ipynb

```diff
@@ -0,0 +1,1434 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Data Exploration \n",
+    "Data exploration involves summarising the dataset's main characteristics and visualising its distribution and relationships. This step helps identify patterns, trends, and potential issues such as missing values, setting foundation for subsequent analysis and modelling."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Import Libraries\n",
+    "Essential libraries such as `pandas`, `os`, and visualisation libraries like `matplotlib` are imported. These libraries provide the necessary functions to handle and visualise the data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 52,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import matplotlib.pyplot as plt\n",
+    "import seaborn as sns\n",
+    "from sklearn.preprocessing import LabelEncoder\n",
+    "import os"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Loading dataset\n",
+    "The dataset is loaded from a csv file from the `datasets` folder into a DataFrame using `pandas`. This allows for easy manipulation and analysis of data."
```

Comment:
```suggestion
    "The dataset is loaded from a CSV file from the `datasets` folder into a `pandas` `DataFrame`."
```

Don't think we need that last sentence since it's fairly self-explanatory.